### PR TITLE
EDUCATOR-4726 -  show file name and description

### PR DESCRIPTION
**Pull request by Andytr1**
This is a WIP - to see if this approach is sufficient for our needs.

### CODE REVIEW

File: openassessment/xblock/static/js/src/lms/oa_response.js [hunk 1]
```diff
@@ -748,8 +749,13 @@ OpenAssessment.ResponseView.prototype = {
...
         var sel = $('.step--response', this.element);
-
-        return this.server.saveFilesDescriptions(this.filesDescriptions).done(
+        var descriptionsAndNamesArray = [];
+        for (var i=0; i < this.filesDescriptions.length; i++) {
+            this.fileNames.push(this.files[i].name);
+            var entry ={description:this.filesDescriptions[i], fileName: this.files[i].name };
+            descriptionsAndNamesArray.push(entry);
+        }
+        return this.server.saveFilesDescriptions(descriptionsAndNamesArray).done(
```
[the remark this answers is not in the document]
Comment (iloveagent57) [reply openassessment/xblock/static/js/src/lms/oa_response.js]: You should also modify that function to structure the JSON data in an easier to understand way.  Right now, it's got a key called `descriptions` which points at a list of dicts that each contain a `description` and `fileName` key.  Maybe change that top-level `descriptions` key to `fileMetadata`?

File: openassessment/xblock/submission_mixin.py [hunk 2]
```diff
@@ -203,19 +206,22 @@ def save_files_descriptions(self, data, suffix=''):  # pylint: disable=unused-ar
...
 
         Returns:
             dict: Contains a bool 'success' and unicode string 'msg'.
         """
         if 'descriptions' in data:
-            descriptions = data['descriptions']
+            file_data = data['descriptions']
 
-            if isinstance(descriptions, list) and all([isinstance(description, six.string_types) for description in descriptions]):
+            if isinstance(file_data, list) and all([isinstance(description['description'], six.string_types) and
```
Comment (iloveagent57) [inline openassessment/xblock/submission_mixin.py]: Nit: you can break this across lines in a more readable way
```python
if isinstance(file_data, list) and all([
    isinstance(description['description'], six.string_types) and
    isinstance(description['fileName'], six.string_types)
    for description in file_data
]):
```

File: openassessment/xblock/static/js/src/lms/oa_response.js [hunk 3]
```diff
@@ -748,8 +749,13 @@ OpenAssessment.ResponseView.prototype = {
     saveFilesDescriptions: function() {
         var view = this;
         var sel = $('.step--response', this.element);
-
-        return this.server.saveFilesDescriptions(this.filesDescriptions).done(
+        var descriptionsAndNamesArray = [];
+        for (var i=0; i < this.filesDescriptions.length; i++) {
+            this.fileNames.push(this.files[i].name);
```
[the remark this answers is not in the document]
Comment (Andytr1) [reply openassessment/xblock/static/js/src/lms/oa_response.js]: checkFilesDescriptions is called anytime there are changes in the description text box. I would like to keep it here, as it's done immediately before we send the information to the server.

File: openassessment/xblock/static/js/src/lms/oa_response.js [hunk 4]
```diff
@@ -748,8 +749,13 @@ OpenAssessment.ResponseView.prototype = {
     saveFilesDescriptions: function() {
         var view = this;
         var sel = $('.step--response', this.element);
-
-        return this.server.saveFilesDescriptions(this.filesDescriptions).done(
+        var descriptionsAndNamesArray = [];
```
[the remark this answers is not in the document]
Comment (Andytr1) [reply openassessment/xblock/static/js/src/lms/oa_response.js]: done

File: openassessment/xblock/static/js/src/lms/oa_response.js [hunk 5]
```diff
@@ -748,8 +749,13 @@ OpenAssessment.ResponseView.prototype = {
...
+        for (var i=0; i < this.filesDescriptions.length; i++) {
```
[the remark this answers is not in the document]
Comment (Andytr1) [reply openassessment/xblock/static/js/src/lms/oa_response.js]: Lets chat

File: openassessment/xblock/submission_mixin.py [hunk 6]
```diff
@@ -110,12 +110,15 @@ def submit(self, data, suffix=''):  # pylint: disable=unused-argument
...
+                    saved_files_names = json.loads(self.saved_files_names)
                 except ValueError:
                     saved_files_descriptions = None
+                    saved_files_names = None
                 submission = self.create_submission(
                     student_item_dict,
                     student_sub_data,
-                    saved_files_descriptions
+                    saved_files_descriptions,
+                    saved_files_names
```
[the remark this answers is not in the document]
Comment (Andytr1) [reply openassessment/xblock/submission_mixin.py]: done

File: openassessment/xblock/submission_mixin.py (hunk 2 above)
Comment (Andytr1) [reply openassessment/xblock/submission_mixin.py]: done

File: openassessment/xblock/static/js/src/lms/oa_response.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Andytr1) [reply openassessment/xblock/static/js/src/lms/oa_response.js]: done

File: openassessment/xblock/static/js/src/lms/oa_response.js [hunk 7]
```diff
@@ -748,8 +749,13 @@ OpenAssessment.ResponseView.prototype = {
...
+            this.fileNames.push(this.files[i].name);
+            var entry ={description:this.filesDescriptions[i], fileName: this.files[i].name };
```
[the remark this answers is not in the document]
Comment (Andytr1) [reply openassessment/xblock/static/js/src/lms/oa_response.js]: done

File: openassessment/xblock/static/js/spec/oa_server.js [hunk 8]
```diff
@@ -187,7 +187,7 @@ describe("OpenAssessment.Server", function() {
         expect($.ajax).toHaveBeenCalledWith({
             url: "/save_files_descriptions",
             type: "POST",
-            data: JSON.stringify({descriptions: ['test1', 'test2']}),
+            data: JSON.stringify({fileMetadata: ['test1', 'test2']}),
```
Comment (iloveagent57) [inline openassessment/xblock/static/js/spec/oa_server.js]: Can we add a test that indicates that file names are also submitted in this POST request?

File: openassessment/xblock/submission_mixin.py [hunk 9]
```diff
@@ -260,12 +272,14 @@ def create_submission(self, student_item_dict, student_sub_data, files_descripti
                         "descriptions {files_descriptions}".format(
                             student_item_dict=student_item_dict,
                             student_sub_data=student_sub_data,
-                            files_descriptions=files_descriptions 
+                            files_descriptions=files_descriptions,
+                            files_names=files_names,
```
Comment (Ayub-Khan) [inline openassessment/xblock/submission_mixin.py]: FYI: Not used.

Comment (Ayub-Khan) [reply openassessment/xblock/submission_mixin.py]: Also, I think this is called for every single file so this should log for one file. My assumption is that this is printing same exception for all FileUploadErrors.

**Pull request merged**